sync-core/src/main/java/com/cloudant/sync/datastore/migrations/MigrateDatabase6To7.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2015 IBM Cloudant, Inc. All rights reserved.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.sync.datastore.migrations;
+
+import com.cloudant.sync.sqlite.ContentValues;
+import com.cloudant.sync.sqlite.Cursor;
+import com.cloudant.sync.sqlite.SQLDatabase;
+import com.cloudant.sync.sqlite.SQLQueueCallable;
+
+import org.apache.commons.codec.binary.Hex;
+
+/**
+ * Migration from datastore JSON database version 6 to 7, which requires migration of data
+ * between tables with a transform in Java code along the way.
+ */
+public class MigrateDatabase6To7 implements Migration {
+
+    public void runMigration(SQLDatabase db) throws Exception {
+
+        db.execSQL("CREATE TABLE attachments_key_filename ( " +
```

Comment:
It really depends. If I did that, the DatastoreConstants would appear to contain the migration because it had the schema, but in fact it wouldn't have the full story of what happened to the database.

But happy to move if this argument doesn't sway you.